src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Invariant.cs

```diff
@@ -66,10 +66,11 @@ private static void InvariantCreateSortKeyOrdinalIgnoreCase(ReadOnlySpan<char> s
                     if (char.IsLowSurrogate(cl))
                     {
                         SurrogateCasing.ToUpper(c, cl, out char hr, out char lr);
-                        BinaryPrimitives.WriteUInt16BigEndian(sortKey, hr);
-                        BinaryPrimitives.WriteUInt16BigEndian(sortKey, lr);
-                        i++;
+                        Span<byte> tmp = sortKey.Slice(0, 2 * sizeof(ushort)); // help with bounds check elimination
```

Comment:
Why is this needed? Why is it not enough to use the SortKey span directly?

Reply:
By slicing the span before writing, we can avoid two bounds checks:

https://www.diffchecker.com/UFgsRYlS/

[godbolt](https://csharp.godbolt.org/z/GvfWsoocs)